### ISSUE TITLE
DropDown/Spinner does not toggle if added to RelativeLayout (if layout pos != 0, 0)

### ISSUE THREAD

**Issue by werton** [kivy/kivy#4092]
on_touch_down() method of DropDown widget not properly handling touch event if self.attach_to attached to relativelayout. I.e. self.attach_to has relative coordinates, but touch.pos has absolute positioning. touch.pos should be translated to_widget position, otherwise there will be unnecessary call self.dismiss().
the problem this piece of code:

```
    def on_touch_down(self, touch):
        if super(DropDown, self).on_touch_down(touch):
            return True
        if self.collide_point(*touch.pos):
            return True
        if self.attach_to and self.attach_to.collide_point(*touch.pos):  # touch.pos should be translated to_widget position
            return True
        if self.auto_dismiss:
            self.dismiss()
```
Here runnable example, the Button 1 is attached to RelativeLayout and can't be toggled as opposed to Button 2 which not attached to RelativeLayout

```
from kivy.lang import Builder
from kivy.base import runTouchApp

w = Builder.load_string('''
Widget:
    RelativeLayout:
        pos: 0, 200

        Button:
            id: btn1
            pos: 0,0
            text: 'Button 1'
            on_release: dropdown.open(self)

    Button:
        id: btn2
        pos: 200,0
        text: 'Button 2'
        on_release: dropdown.open(self)

    DropDown:
        id: dropdown
        Button:
            text: 'Value A'
            size_hint_y: None
            on_release: dropdown.select('A')

        Button:
            text: 'Value B'
            size_hint_y: None
            on_release: dropdown.select('B')

''')

runTouchApp(w)
```

**Comment by werton** [comment]
Updated example. In it two buttons, one button is placed on RelaytiveLayout. If you click the buttons you will see that dropdown's button1 is not correctly handling the event on_touch_down(i.e. dropdown cannot be closed by clicking on
button after opening), it occurs in consequence of the fact that on_touch_down method does not translate the coordinates of button 1 to the absolute coordinates.

```
from kivy.base import runTouchApp
from kivy.lang import Builder
from kivy.uix.widget import Widget

Builder.load_string('''
<MyWidget>:
    # This is a debug rendering dropdown's button position which received dropdown.on_touch_down method
    touch_coordinates: [0, 0]
    dropdown_button_coord: [0, 0, 0, 0]

    canvas.after:
        Color:
            rgba: 0, 1, 0, 0.5
        Rectangle:
            pos: self.dropdown_button_coord[:2]
            size: self.dropdown_button_coord[2:]

        Color:
            rgba: 1, 0, 0, 1
        Rectangle:
            pos: [x - 5 for x in self.touch_coordinates]
            size: 10, 10

    on_touch_down:
        print('touch.pos: ', self.touch_coordinates)
        print('dropdown\\'s button coordinates: ', self.dropdown_button_coord[:2])

    # Adds two buttons. The Button 1 placed on RelativeLayout.
    # RelativeLayout displaced along the x,y-axis to create an offset between the
    # relative coordinates of the RelativeLayout and the absolute coordinates.
    RelativeLayout:
        pos: 100, 200

        Button:
            id: btn1
            pos: 0,0
            text: 'Button 1'
            on_release: dropdown.open(self)

    Button:
        id: btn2
        pos: 300, 200
        text: 'Button 2'
        on_release: dropdown.open(self)


    DropDown:
        id: dropdown
        # To avoid display the dropdown at the program start set the x to -100
        pos: -100, 0

        # Update the saved dropdown button coordinates for debug rendering
        on_touch_down: root.update_draw_positions(args[1], self)

        Button:
            text: 'Value A'
            size_hint_y: None
            on_release: dropdown.select('A')

        Button:
            text: 'Value B'
            size_hint_y: None
            on_release: dropdown.select('B')
''')


class MyWidget(Widget):

    def update_draw_positions(self, touch, dropdown):
        self.touch_coordinates = touch.pos
        if dropdown.attach_to:
            self.dropdown_button_coord = [dropdown.attach_to.x, dropdown.attach_to.y, dropdown.attach_to.width, dropdown.attach_to.height]
        self.canvas.ask_update()

runTouchApp(MyWidget())
```

**Comment by werton** [comment]
Since the Spinner uses by default the DropDown class as dropdown_cls, this bug manifests itself in it.

```
from kivy.base import runTouchApp
from kivy.lang import Builder
from kivy.uix.widget import Widget

Builder.load_string('''
<MyWidget>:
    # Adds two Spinner. The Spinner 1 placed on RelativeLayout.
    # RelativeLayout displaced along the x,y-axis to create an offset between the
    # relative coordinates of the RelativeLayout and the absolute coordinates.
    RelativeLayout:
        pos: 100, 200

        Spinner:
            id: spinner1
            text: 'spinner1'
            values: ('Home', 'Work', 'Other', 'Custom')
            pos: 0,0

    Spinner:
        id: spinner2
        text: 'spinner2'
        values: ('Home', 'Work', 'Other', 'Custom')
        pos: 300, 200
''')

class MyWidget(Widget):
    pass
runTouchApp(MyWidget())
```

**Comment by akshayaurora** [comment]
closed by #4112
